roottest/root/io/datamodelevolution/misc/CMakeLists.txt

```diff
@@ -1,7 +1,54 @@
-#-------------------------------------------------------------------------------
-#
-# Placeholder file to translate the tests to the new CTest system. Meanwhile we
-# define a CTest test that calls 'make' in ${CMAKE_CURRENT_SOURCE_DIR}
-#
-#-------------------------------------------------------------------------------
-ROOTTEST_ADD_OLDTEST(LABELS longtest)
+ROOTTEST_GENERATE_DICTIONARY(libArrayRules_dictcint
+                  arrayRules.h
+                  LINKDEF arrayRulesLinkDef.h
+                  NO_ROOTMAP NO_CXXMODULE)
+
+ROOTTEST_GENERATE_REFLEX_DICTIONARY(libArrayRules_dictrflx
+                  arrayRules.h
+                  SELECTION arrayRulesSelection.xml
+                  LIBNAME libArrayRules_dictrflx
+                  NO_ROOTMAP)
+
+ROOTTEST_ADD_TEST(Conversions
+                  MACRO execConversions.cxx+
+                  OUTREF execConversions.ref
+                  LABELS longtest)
+
+ROOTTEST_ADD_TEST(WriteOrderClass
+                  MACRO execWriteOrderClass.cxx+
+                  OUTREF execWriteOrderClass.ref
+                  LABELS longtest
+                  FIXTURES_SETUP WriteOrderClass)
+
+ROOTTEST_ADD_TEST(ReadOrderClass
+                  MACRO execReadOrderClass.cxx+
+                  OUTREF execReadOrderClass.ref
+                  LABELS longtest
+                  FIXTURES_REQUIRED WriteOrderClass)
+
+ROOTTEST_ADD_TEST(BuildMyClass
+                  MACRO ${CMAKE_SOURCE_DIR}/roottest/scripts/build.C
+                  MACROARG "\"MyClass.cxx\""
+                  COPY_TO_BUILDDIR MyClass.h MyClass.cxx MyClass_linkdef.h
+                  FIXTURES_SETUP BuildMyClass-fixture)
+
+ROOTTEST_ADD_TEST(BuildSomeClass
```

Comment:
Is this better than `ROOTTEST_COMPILE_MACRO` ?

Reply:
Yes, I replace it here